src/backend/distributed/executor/local_executor.c

```diff
@@ -585,6 +588,39 @@ ExecuteLocalTaskPlan(PlannedStmt *taskPlan, char *queryString,
 }
 
 
+/*
+ * RecordReferenceTableAccessesForTask records relation accesses for reference
+ * tables that given task will access.
+ */
+static void
+RecordReferenceTableAccessesForTask(Task *task)
+{
+	ShardPlacement *taskPlacement = NULL;
+	foreach_ptr(taskPlacement, task->taskPlacementList)
+	{
+		uint64 shardId = taskPlacement->shardId;
+		if (shardId == INVALID_SHARD_ID)
+		{
+			/*
+			 * When a SELECT prunes down to 0 shard, we still may pass through
+			 * the local executor. In that case, we don't need to record any
+			 * relation access as we don't actually access any shard placement.
+			 */
+			continue;
+		}
+
+		Oid relationId = RelationIdForShard(shardId);
```

Comment:
One improvement could be add an early return here:

```C
if (CitusTableTtype(relationId, NO_KEY)
return
```

Also, it'd be easier to  use the naming of `NoDistKey` to this new function, to align the local tables PR

Reply:
>Also, it'd be easier to use the naming of NoDistKey

I was planning to rename the underlying function the new funtction to use `NoDistKey` in planner pr for completeness, does that sound good 

>One improvement could be add an early return here:

👍

Reply:
>One improvement could be add an early return here:

edit:
I couldn't get the suggestion here, why do we want to early return for `DIST_BY_NONE` tables here ?